app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterClaimController.scala

```diff
@@ -79,76 +78,86 @@ class EnterClaimController @Inject() (
             logger.warn("No duties found to create claims ")
             Redirect(routes.SelectDutiesController.selectDuties())
           case Right(Some(claims)) =>
-            val nextPage = calculateNextPage(claims)
+            val nextPage = calculateNextPage(claims, journeyBindable)
             updateClaimAnswer(claims, fillingOutClaim, nextPage)
         }
       }
     }
 
-  def enterClaim(id: UUID): Action[AnyContent] =
+  def enterClaim(taxCategory: TaxCategory, taxCode: TaxCode, journeyBindable: JourneyBindable): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withAnswers[ClaimsAnswer] { (fillingOutClaim, answer) =>
-        answer.flatMap(_.find(_.id === id)) match {
+      implicit val journey: JourneyBindable = journeyBindable
```

Comment:
This looks a like a design smell. I looked at the `withAnswersAndRoutes` definition and you have designed it to take an implicit journeyBindable. Correct me if I am wrong but that is explicitly set in the routes file. Therefore the controller method would take a non-implicit parameter of the same type which could be passed into your `withAnswersAndRoutes`. The point is that you have gained nothing here making it `withAnswersAndRoutes` take an implicit journeyBindabe, unless I glaringly missing something obvious. We can talk about it on a call if you like.